pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -144,6 +146,19 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 			controlPlaneDeploymentNeeded, err = needsControlPlaneDeployment(ctx, o, kubeAPIServerDeploymentFound, infrastructure)
 			return err
 		}),
+		errors.ToExecute("Check if there are DNSEntry resources in the shoot namespace in the seed cluster", func() error {
+			dnsEntryList := &dnsv1alpha1.DNSEntryList{}
+			if err := botanist.K8sSeedClient.APIReader().List(ctx, dnsEntryList, client.InNamespace(o.Shoot.SeedNamespace)); err != nil {
+				return err
+			}
+			for _, dnsEntry := range dnsEntryList.Items {
+				if len(dnsEntry.Status.Targets) > 0 {
```

Comment:
Ideally, we would check if the `DNSEntry` has any associated `DNSProvider` and only exclude those from `deployAdditionalDNSProviders` which don't manage any `DNSEntries`.

The reasons are:
- There will always be `DNSEntries` if the feature gate `UseDNSRecords` is off.
- One can have multiple provider under which only one or a few have an issue.

Reply:
Hm, yes, you are right. This would make the check a bit more complex though. Let me have another try.

Reply:
Changed as suggested, PTAL.